.github/workflows/docs.yml

```
@@ -0,0 +1,45 @@
+name: Publish Docs
+
+on:
+  workflow_dispatch: # allow manual runs
+  push:
+    branches:
+      - main
+    paths: # avoid unnecessary builds
+      - docs/**
+      - book.toml
+      - .github/workflows/docs.yml
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: github-pages
+  cancel-in-progress: false # skip any intermediate builds but let finish
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      MDBOOK_VERSION: 0.4.40
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install mdBook
+        run: cargo install --version ${MDBOOK_VERSION} mdbook
```

OR! This can be changed to binaries to cut the build time from minutes to mere seconds:

```suggestion
        run: curl -sSL "https://github.com/rust-lang/mdBook/releases/download/v$MDBOOK_VERSION/mdbook-v$MDBOOK_VERSION-x86_64-unknown-linux-gnu.tar.gz" | tar -xz --directory /home/runner/.cargo/bin
```

(it's ugly, seems fragile, but when it's working for autopush-rs…;)))

---

Lets roll with the slower `cargo` version before we optimize for `curl | tar` hijinks.